tests/test-cases/Group1-Docker-Commands/1-14-Docker-Kill.robot

```diff
@@ -77,12 +83,12 @@ Signal a container with SIGHUP
     Should Be Equal As Integers  ${rc}  1
     ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} kill -s HUP ${id}
     Should Be Equal As Integers  ${rc}  0
-    Wait Until Keyword Succeeds  20x  200 milliseconds  Assert Container Output  ${id}  KillSignalHUP
+    Wait Until Keyword Succeeds  20x  200 milliseconds  Assert Not In Container Output  ${id}  KillSignalHUP
```

Comment:
There's no point in `Wait Until ...` here - that was present previously to wait for expected output. With the inversion of the test this will either immediately succeed or always fail.

That said I'm not sure this test should have it inverted.... I thought we were still wanting to deliver the signal to the main process, which should be the shell running the trap command in this case? Was this supposed to be changed in the test `Confirm signal is not delivered to entire process group`?

Reply:
The purpose of this test is to make sure that the child did not receive the signal. We could assert that the parent gets the signal, but there is an existing test (that does not use the nested shell) just above this one that does exactly that. I am not sure why we shouldn't wait for the output to drain; that is what makes sure that the child process did not receive the signal and processed it by writing the message to the log. This test is meant to catch the error that caused issue: #8152.